metamask.html

```diff
@@ -149,39 +149,52 @@ <h2>Interact with the Contract</h2>
 
 function generateNonce(id) {
   var challenge;
-    this.myContractInstance.currentChallenge.call(function(err, res) {
-      challenge = res;
-      var difficulty;
-      if (id == 'getPaidNonce') {
-        this.myContractInstance.paydayDifficulty.call(
-          function(err, res){
-            difficulty = res;
-            console.log(difficulty);
-            console.log(challenge);
-            var nonce = generateNonceFrom(difficulty, challenge);
-            document.getElementById(id).innerHTML = nonce;
-          }
-        );
-      } else if (id == 'takeFromRichNonce') {
-        this.myContractInstance.robberyDifficulty.call(
-          function(err, res){
-            difficulty = res;
-            console.log(difficulty);
-            console.log(challenge);
-            var nonce = generateNonceFrom(difficulty, challenge);
-            document.getElementById(id).innerHTML = nonce;
-          }
-        );
-      } else {
-        return;
-      }
-    });
+  this.myContractInstance.currentChallenge.call(function(err, res) {
+    challenge = res;
+    if (id == 'getPaidNonce') {
+      this.myContractInstance.paydayDifficulty.call(
+        function(err, res){
+          var difficulty = '0x' + res.toString(16);
+          var nonce = generateNonceFrom(difficulty, challenge);
+          document.getElementById(id).value = nonce;
+        }
+      );
+    } else if (id == 'takeFromRichNonce') {
+      this.myContractInstance.robberyDifficulty.call(
+        function(err, res){
+          var difficulty = '0x' + res.toString(16);
+          var nonce = generateNonceFrom(difficulty, challenge);
+          document.getElementById(id).value = nonce;
+        }
+      );
+    } else {
+      return;
+    }
+  });
 }
 
-function generateNonceFrom(difficulty, challenge) {
-    var nonce = 'nonce';
-
-    return nonce;
+function generateNonceFrom(target, challenge) {
+  let nonce = 0;
+  let done = false;
+
+  console.log('starting to mine');
+  console.log('target: ', target);
+  console.log('challenge: ', challenge);
+
+  // TODO: Run is a web worker
+  while (!done) {
+    let data = '0x' + nonce.toString(16) + challenge.replace('0x', '')
```

Comment:
I think there is something wrong here

Reply:
do you mean to use `target`? also you change the target.toString(16) just before you pass it into this function. no need to do it twice